src/controls/polyControl.test.ts

```diff
@@ -0,0 +1,116 @@
+import { Point, Polyline, type Transform } from 'fabric';
+import { degreesToRadians, saveObjectTransform } from '../util';
+import {
+  createPolyControls,
+  factoryPolyActionHandler,
+  polyActionHandler,
+} from './polyControl';
+import { CENTER } from '../constants';
+
+//
+// prep for tests
+//
+const createPolylineMovementItems = (points: Point[], pointDiff: Point) => {
+  // create polyline with controls
+  const target = new Polyline(points, {
+    stroke: 'black',
+    strokeWidth: 0,
+    originX: 'center',
+    originY: 'center',
+    strokeUniform: false,
+    exactBoundingBox: false,
+  });
+  const controls = createPolyControls(target);
+  target.controls = controls;
+  const corner = 'p1';
+  // create mouse pointer that moves the second point slightly
+  const pointer = new Point(target.points[1]).subtract(pointDiff);
+
+  // create transform event based on polyline
+  const transform: Transform & { pointIndex: number } = {
```

Comment:
does `canvas._setupCurrentTransform` instead do the job?
Maybe then assigning some values to it or using jest spies.

Reply:
Pulling out code to isolate problem. Separate PR can be done for polycontrol issue.